### PR TITLE
enable tmath tests for JS backend

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: tests/stdlib/tmath.nim [hunk 1]
```diff
@@ -1,13 +1,12 @@
 discard """
-  action: run
-  matrix:"; -d:nimTmathCase2 -d:danger --passc:-ffast-math"
+  targets: "c js"
 """
 
-# xxx: fix bugs for js then add: targets:"c js"
+## xxx enable matrix:"; -d:nimTmathCase2 -d:danger --passc:-ffast-math"
```
Comment (Araq) [inline tests/stdlib/tmath.nim:5]: In the worst case, create a tmath2.nim test that has this test matrix and includes tmath.nim

Comment (ringabout) [reply tests/stdlib/tmath.nim:5]: the test even doesn't work locally. I don't understand why they can pass the tests before
`nim_temp.exe  c -r -d:nimTmathCase2 -d:danger --passc:-ffast-math tests/stdlib/tmath.nim`

```
fatal.nim(49)            sysFatal
Error: unhandled exception: tmath.nim(204, 15) `classify(trunc(f_neg_one * f_zero)) == fcNegZero`  [AssertionDefect]
```

File: tests/stdlib/tmath.nim [hunk 2]
```diff
@@ -1,13 +1,12 @@
 discard """
-  action: run
-  matrix:"; -d:nimTmathCase2 -d:danger --passc:-ffast-math"
+  targets: "c js"
```
Comment (timotheecour) [inline tests/stdlib/tmath.nim]:
```suggestion
  targets: "c cpp js"
```

just to be on the safe side; cpp is usually redundant with c but math is important enough

File: tests/stdlib/tmath.nim (hunk 1 above)
Comment (ringabout) [reply tests/stdlib/tmath.nim:5]: @Araq
It never works(there is a serious bug in testament)
see https://github.com/nim-lang/Nim/pull/16567#issuecomment-753681785

**Review by timotheecour — APPROVED** [review]
LGTM
indeed `--passc:-ffast-math` seems to only work in CI because of a bug in testament (https://github.com/nim-lang/Nim/pull/16567#issuecomment-753681785); note that it does seem to work locally for some OS/compilers (eg locally on OSX for me), see https://stackoverflow.com/questions/22931147/stdisinf-does-not-work-with-ffast-math-how-to-check-for-infinity/22931368#22931368 but it may as well be just considered subject to implementation details

**Pull request merged**